examples/nodes/torchata_nodes_basics.ipynb

```diff
@@ -0,0 +1,439 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b517ebcf-9f92-43bf-b332-3875933ea244",
+   "metadata": {},
+   "source": [
+    "#### Batch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6156bcb4-891f-484a-bf06-cbf39e282175",
+   "metadata": {},
+   "source": [
+    "A BaseNode can be passed into a Batcher, to get batches of size `batch_size`.\n",
+    "By default, drop_last is True, meaning if the last batch has a size smaller than the `batch_size`, it is not produced."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "dc40c637-90ed-4c09-9961-3b0d488fc6d8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0, 1, 2, 3]\n",
+      "[4, 5, 6, 7]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from torchdata.nodes import Batcher\n",
+    "batched = Batcher(source, batch_size = 4)\n",
+    "for batch in batched:\n",
+    "    print(batch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5487e01d-6245-4940-a0df-c32aee0849cf",
+   "metadata": {},
+   "source": [
+    "We can make `drop_last = False` to produce the last batch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "092d8e5b-aeb2-4de1-9b13-8b0dc0af31ce",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0, 1, 2, 3]\n",
+      "[4, 5, 6, 7]\n",
+      "[8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batcher = Batcher(source, batch_size = 4, drop_last = False)\n",
+    "for batch in batcher:\n",
+    "    print(batch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e11543b9-c2dd-4861-af24-0eb3a1f3ddb9",
+   "metadata": {},
+   "source": [
+    "If we try to use this batcher over multiple epochs, we will need to reset it after every epoch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "d89e36a9-cc9c-4b55-acba-dbcaf8d53407",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "On epoch = 0\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n",
+      "On epoch = 1\n",
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batcher = Batcher(source, batch_size = 10)\n",
+    "num_epochs = 2\n",
+    "\n",
+    "for epoch in range(num_epochs):\n",
+    "    print(f\"On epoch = {epoch}\")\n",
+    "    for batch in batcher:\n",
+    "        print(batch)\n",
+    "    batcher.reset()\n",
+    "    \n",
+    "# This is one extra step than traditional dataloader, we can actually wrap the batcher in a Loader to skip that\n",
+    "# Let's look at Loader in the next cell"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d2232823-0927-432b-9171-fb2bf38a7205",
+   "metadata": {},
+   "source": [
+    "#### Loader\n",
+    "As you can see, we get a batch in every epoch, without even needing to reset the loader!!"
```

Comment:
This would be a good opportunity to showcase SamplerWrapper, and how it will increment epoch automatically, can you maybe add a new cell above here to demonstrate SamplerWrapper, and then how it works together with Loader? We could print the epoch value of eg RandomSampler, and show the different results after shuffling

Reply:
Sure.